### ISSUE TITLE
feat: ONNXMlir first class support

### ISSUE THREAD

**Issue by aarnphm** [bentoml/BentoML#2693]
Provide supports for ONNXMlir as a first-class citizen

TODO:
- [ ] use the docker container as an entrypoint for `onnx-mlir`
- [ ] provides envar `ONNX_MLIR_DIR` if users build ONNX_MLIR locally
- [ ] need a way to safely add PyRuntime to `sys.path`
- [ ] support `multi_entrypoint` compiled model.

Maybe provides support for older `ExecutionSession` for older onnx-mlir API. Since this library is pretty obscure I assume everyone are probably living on the edge and install the latest changes anyway.

This will be looked after 1.0 releases, as ONNXMlir is not a top list item.

**Comment by messerb5467** [comment]
@andrewsi-z for your awareness

**Comment by andrewsi-z** [comment]
Sounds good! 
@aarnphm with 'use the docker container as an entrypoint for onnx-mlir' I assume this would be to take an ONNX model as input and handle compile as part of the abstraction layer rather than require the compile as a prerequisite step (which the current approach relies on)?

**Comment by aarnphm** [comment]
that is correct. We will allow any .onnx file format and we will use the docker container to compile the model.

**Comment by aarnphm** [comment]
I'm thinking to also provide a helper layer where users can put any model file (let say tensorflow model format, or torch model format) and then convert it to onnx. But imo that is outside the scope abit. We could do something like that for ONNX as well in the future.

**Comment by larme** [comment]
@aarnphm I wonder if we want to convert tf/pt model to onnx for users, there're a lot of fine tuning can happen in this process. Better give user some references in documentations
